### DISCUSSION TITLE
Implementing an I/O interface.

### DISCUSSION THREAD

**Discussion by Noobie2014** [hneemann/Digital#1349] [Q&A] [answered]
How is a bidirectional tri-state interface created without the simulator crashing to the 2 outputs connected together error?
doesn't the synthesizer realize that using an inverted enabled tri-sate and an non inverted enabled tri-state will not have both
on at the same time? just having an input and output component doesn't allow me to use bidirectional logic. Please tell me
how I can create this component. Even the 74245 transceiver doesn't simulate.

**Comment by hneemann** [comment]
Sometimes users do not realize that an input outputs a value. It is always connected to VCC or GND. If it is not supposed to do this, it must output "high-z", which means that nothing is connected to it. The input can be configured to output “high-z”.
I suspect that your problems are due to the fact that you are connecting two inputs together via the buffers, and that it has nothing to do with the buffers themselves.

**Reply by Noobie2014** [reply]
I get that the input gizmo actually outputs and the output actually gets an input as devices that generate an input and receive an output but what do you mean about they always connected to power or ground? Inputs can always be connected together through diode logic for one, I see that its actually a diode OR gate but still the inputs are connected.  What I was asking is how would you implement a GPIO pin using the gizmos you supply? your input and output have no dynamic switchable states that I can see?

Thank You for answering also :)

**Reply by hneemann** [reply]
I'm not quite sure what the question is, but no special logic is needed to switch the direction of inputs or outputs. If subcircuits are connected together, inputs and outputs are bidirectional.

**Comment by Noobie2014** [comment, marked as answer]
Thank you, My question was how would I implement an interface that turns into an output or input according to a directional signal or bit. Look how peripheral ports operate. I am trying to use your imbeded circuit to do this. I thank you for trying to help. I'll just breadboard a circuit to figure this out.

Thank You

Yahoo Mail: Search, Organize, Conquer 
 
  On Mon, Oct 21, 2024 at 8:26 AM, Helmut ***@***.***> wrote:   


I'm not quite sure what the question is, but no special logic is needed to switch the direction of inputs or outputs. If subcircuits are connected together, inputs and outputs are bidirectional.
